core/src/main/scala/org/apache/spark/sql/rapids/tool/ToolUtils.scala

```diff
@@ -53,10 +55,35 @@ object ToolUtils extends Logging {
     // case class to hold key -> value pairs
     case class ClusterTags(key: String, value: String)
     implicit val formats = DefaultFormats
-    val listOfClusterTags = parse(clusterTag)
-    val clusterTagsMap = listOfClusterTags.extract[List[ClusterTags]].map(
-      x => x.key -> x.value).toMap
-    clusterTagsMap
+    try {
+      val listOfClusterTags = parse(clusterTag)
+      val clusterTagsMap = listOfClusterTags.extract[List[ClusterTags]].map(
+        x => x.key -> x.value).toMap
+      clusterTagsMap
+    } catch {
+      case NonFatal(_) =>
+        logWarning(s"There was an exception parsing cluster tags string: $clusterTag, skipping")
+        Map.empty
+    }
+  }
+
+  /**
+   * Try to get the JobId from the cluster name. Parse the clusterName string which
+   * looks like:
+   * "spark.databricks.clusterUsageTags.clusterName":"job-557875349296715-run-4214311276"
+   * and look for job-XXXXX where XXXXX represents the JobId.
+   *
+   * @param clusterNameString String which contains property clusterUsageTags.clusterName
+   * @return Optional JobId if found
+   */
+  def parseClusterNameForJobId(clusterNameString: String): Option[String] = {
+    var jobId: Option[String] = None
+    val splitArr = clusterNameString.split("-")
+    if (splitArr.contains("job")) {
+      val jobIdx = splitArr.indexOf("job")
+      jobId = Some(splitArr(jobIdx + 1))
+    }
```

Comment:
This block could cause an exception for strings with `jobIdx == splitArr.length - 1`
If this never happens in real world then we can keep the code as is.
Otherwise, we can add try/catch or check the jobIdx before the assignment.
```
val jobIdx = splitArr.indexOf("job") + 1
if (jobIdx > 0 && jobIdx < splitArr.len()) {
  jobId = Some(splitArr(jobIdx))
}
```

Reply:
its possible only if the string is not formatted like expected. so if it had only "job" in it rather then "job-XXXX", I'll add a check just in case

Reply:
note you don't need jobIdx > 0 check because we already verified that splitArr contains job

Reply:
Yes, that's true. I assumed the if-condition  `splitArr.contains("job")` would be removed, while relying on `indexOf`.